product_assortment/models/ir_filters.py

```diff
@@ -64,7 +64,10 @@ def _get_eval_domain(self):
 
         return res
 
-    @api.depends("domain", "blacklist_product_ids", "whitelist_product_ids")
+    # Original : @api.depends("domain", "blacklist_product_ids", "whitelist_product_ids")
+    # TODO : Resolve bug if depending of blacklist_product_ids and
```

Comment:
is this TODO to be solved now?

Reply:
No, it s working without. And sorry about my PR name, it was not very clear what it is about.

Reply:
but these fields are mandatory for computing the domain. Could you be more explicit on the error? The issue you linked does not explain the reason for this.

Reply:
@Kev-Roche

Reply:
It s an Access Error happending when saving a product in black/whitelist:
"Database Fetch misses ids ((<NewID origin=12>,)) and has extra ids ((12,)), may be caused by a type incoherence in a previous request".

Reply:
Hi @Kev-Roche 
I've included your PR in my project and i still have this kind of issue as soon as i use whitelist/blacklist 
Database fetch misses ids ((<NewId origin=49486>,)) and has extra ids ((49486,)), may be caused by a type incoherence in a previous request

Reply:
@Kev-Roche I just push a PR on your repo fixing the problem: https://github.com/akretion/product-attribute/pull/13
The depends is not needed as the field is not stored. If you depends on the domain, the widget "domain" disturb the cache.

Reply:
@Cedric-Pigeon Thank you for your PR and finding the real problem.

Reply:
@Cedric-Pigeon , it's weird, a test failed with your PR

Reply:
@Kev-Roche yes it is only a cache issue. Adding self.assortment.invalidate_cache() before line 106 should fix the test